tempest-internal/src/main/kotlin/app/cash/tempest/internal/Codec.kt

```diff
@@ -95,7 +94,11 @@ internal class ReflectionCodec<A : Any, D : Any> private constructor(
       .map { it.parameter to it.getDb(dbItem) }
       .toMap()
     val appItem =
-      appItemConstructor?.callBy(constructorArgs) ?: appItemClassFactory.newInstance()
+      if (appItemConstructor != null && constructorArgs.size == appItemConstructor.parameters.size) {
```

Comment:
wanna add an explicit check to see if it’s a record? rather than relying on constructor parameter count equaling field count?

Reply:
This is tricky because this gradle module compiles against Java 11 which doesn't have the Record base class